src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -30,15 +31,22 @@ internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
         /// </summary>
         private readonly bool enableNewDiagnosticEvents;
 
+        private readonly TelemetryConfiguration configuration;
         private readonly TelemetryClient client;
         private readonly IApplicationIdProvider applicationIdProvider;
         private readonly string sdkVersion = SdkVersionUtils.GetVersion();
         private readonly bool injectResponseHeaders;
         private readonly bool trackExceptions;
         private readonly bool enableW3CHeaders;
+        private readonly PropertyFetcher httpContextFetcher = new PropertyFetcher("httpContext");
```

Comment:
not nit: add some info about event this fetcher is used for. It's easy to accidentally reuse fetcher for some other event, but leads to errors. Having event name in the fetcher var name gives a hint  (I hope)  to future us that it not to be reused.